### ISSUE TITLE
Create policy regarding ongoing revisions to lessons

### ISSUE THREAD

**Issue by wcaleb** [programminghistorian/jekyll#498]
As part of #475, @fredgibbs @jerielizabeth @acrymble and I discussed the question of how to resolve the project's potential interest in keeping lessons up-to-date and relevant for users with an author's potential interest in having his or her original lesson preserved as it was at the moment of publication. This question is related to the discussions in #417, #304, #410 and #408.

At suggestion of @jerielizabeth, it was agreed that one way to resolve all or most of these issues would be to draft a default policy regarding revisions made to a lesson by people other than the author. Authors would then be given a choice to "opt out" of having their lesson changed our updated. 

Below is a draft of such a policy for discussion. It's probably overly wordy and convoluted, but I intend it as a starting point for discussion.

---

The *Programming Historian* is committed to giving the authors of lessons full and fair credit for the original work they publish on our site. At the same time, the nature of technical tutorials means that instructions in lessons will sometimes "break" or become less user-friendly due to changes in the software, data sources, or programming languages on which the lessons rely. 

Sometimes these developments may require the *Programming Historian* to alert readers to the fact that a lesson is now outdated and no longer being maintained. But as a matter of policy, we also assume that authors publishing on the *Programming Historian* agree to allow future editors or readers to make minor adjustments to lessons when doing so would extend their usability and ensure that instructions and code blocks continue to work. Authors *can* "opt out" of this default agreement, however, by making clear to an editor at the time of the publication that the author does not want the lesson to be updated in the future. (Authors who choose to forego any future revisions should understand, however, that we reserve the right to place a notice to readers at the top of lessons that no longer work well and are not being actively updated.) 

Either way, the *Programming Historian* is committed to maintaining within our GitHub repository the original Markdown file of all lessons as they appeared at the time of publication. At any time, an author can request and receive a link to our repository's history that shows the lesson text at the time of publication. Our editors can also assist authors in preserving their original lesson in other forms as well; for example, authors may wish to print a copy of their original lesson as a PDF or submit a zipped archive of an original lesson and its associated files to a research and data repository such as [Zenodo](http://zenodo.org) where it can receive its own Digital Object Identifier (DOI).

Finally, none of the above should be construed as giving editors a "blank check" to modify an author's lesson however we choose! Our goal in the first instance is to consult authors and include them in discussions about any major, substantive revisions to their work. The above policy is intended to help determine our course of action when an author may, for various reasons, not want to engage in that ongoing consultative process of revision, leaving editors to decide how to ensure that the lesson remains as functional and relevant as possible to our readers.

**Comment by acrymble** [comment]
Thanks @wcaleb that's a great starting point.

Just to note the license (CC-BY) means we cannot prevent others from making substantial derivative works and publishing them elsewhere. However, I believe that anyone who did so (and us doing so on our own site) are obliged to acknowledge that any revisions make the work 'derived from' rather than 'by' the original author. There are some good examples of that distinction here: http://www.gla.ac.uk/media/media_464806_en.pdf

As you note the original is still kept in the repo - though difficult to find.

**Comment by acrymble** [comment]
Having slept on this, I like the idea of communicating more clearly with authors, but I'm still deeply troubled with the idea that we're the only publication I know of that doesn't keep a clear and easy to access record of what was accepted out of the peer review. I know it exists, but it's not accessible and it's not treated like a particularly important object.

One of the issues I think is what people think our purpose is:

1) A journal where people publish digital method, like the Journal of the Association for History and Computing (https://quod.lib.umich.edu/j/jahc/) #417 #410
2) A dynamic and up-to-date site that teaches the latest and most relevant skills, like Code Academy (https://www.codecademy.com/). #297 #304 #408

I've always seen us as number 1, and everything we do on the editorial end points to number 1 up to the point of updating lessons. Our traffic logs suggest most of our users use it as number 2.

As a potential academic historian author operating in a scholarly context, I'd have the following expectations of a publication offering the number 1 service:

* once accepted a version of record (meaning the value-added version made available by the publisher) would be kept available openly and permanently on the website.
* that this would remain available even once more relevant material had been published, the ideas were eventually disproved, or any future editors no longer felt it was the best that could be offered, as is the case with all journal articles - even retracted ones.
* that any changes would be run past me first, and that I would have a right to refuse them or to refuse to engage with these potentially unwanted additional demands on finished work.

If the journal couldn't guarantee that, I wouldn't publish with them or encourage others to do so.

However, I would be happy for the journal to have a spin-off initiative that sought to take advantage of the CC-BY license to create option 2 listed above, on the condition that:

* Any changed version of the lesson did not have my name reattached as the author without acknowledging alterations were not necessarily made by the author and he/she can not be held responsible for any errors.
* There should be a clear link to an easy to find and use version of the original (like a footnote would serve in traditional publishing).

The assumption that authors all contribute because they want to teach people doesn't apply to me. The lesson I was going to submit over the summer was to do with documenting a method I'd been involved with creating as part of a scholarly project, to encourage scholarly reuse. In that context, I have no interest in a permanently up-to-date version. I have an interest in creating a permanent archive of the method as it was published after peer review, so that it can make its way into scholarly discussions and fall into obscurity (without being deleted) when it's no longer relevant.

So I think we can do both number 1 and number 2, but I don't think you can have 2 without 1.

**Comment by acrymble** [comment]
Building on @wcaleb's text, I wonder if a table is helpful:

| Author decision | Minor issue | Moderate issue | Major issue |
| --------- | --------- | --------- | --------- |
| No changes (leave me alone) | Editor will change anyway | Editor can change, but must do so in a clearly marked aside box that is signed by the editor. | Lesson will be deprecated |
| Consultation only (ok but only if I like them) | Editor will change anyway | Editor can change, but must do so in a clearly marked aside box that is signed by the editor. | Author can choose revision or deprecation, or authorise someone else to do changes. |
| Change away (go for it!) | Editor will change anyway | Editor can change, but must do so in a clearly marked aside box that is signed by the editor. | Editors will change and acknowledge original contribution may differ. |

Minor issues: typos, styling issues, non-substantial broken links, minor wording ambiguity

Moderate issues: reader highlights something that needs clarification, reader highlights a useful tip that would enhance the lesson substantially and that is entirely in scope.

Major issues: underlying technology changes meaning readers now need more direction on dependencies (eg they will likely have installed Python 3 instead of 2), dependencies have broken (data no longer available, service no longer working or updated, beyond everyone's control), major inaccuracies are discovered that are misleading for learners.

**Comment by mdlincoln** [comment]
I'm sorry I had to sign off from the call punctually yesterday, and hence missed this convo. I really appreciate all these ideas, and I agree that conceiving of "versions of record" only makes sense if we sort out these larger questions first. I want to focus in on this, though:

>I've always seen us as number 1, and everything we do on the editorial end points to number 1 up to the point of updating lessons. Our traffic logs suggest most of our users use it as number 2.

I certainly understand that you think of the journal, @acrymble, as being 1 - however, having watched _PH_ from a distance for some time, it seems that absolutely everything this journal has done on developing editorial processes and guidelines, to every way this site presents itself here and on Twitter, points to number **2** instead. Our review processes all emphasize writing a lesson that enhances usability from a pedagogical standpoint, rather than acting as documentation of a _specific_ scholarly process - I mean my goodness, we call these things "lessons" rather than "methodology articles"! We have also recently turned down a lesson proposal precisely because it was _too tightly coupled to a specific research project_, rather than acting as a general tutorial.

So if all this is being driven by the expectation that we are 1, then I simply think there is a mismatch between your own expectations, and the well-planned and successful reality of what _PH_ is?

**Comment by acrymble** [comment]
@mdlincoln you're right, from a user perspective that it's number 2. I suspect (but maybe I'm wrong) that from an author's perspective, some people see it as 1 - as a 'publication'.

I'm writing from the perspective of an author. And when I published my lessons (many years ago now) it was with an expectation that they would be treated as enduring objects in the scholarly conversation. Not something I'd have to defend again later. That's mental energy no one ever is asked to expend on a journal article, no matter how outdated its ideas.

**Comment by jerielizabeth** [comment]
Thank you @acrymble for the thoughtful clarification. It is really helpful for understanding your concerns around updates and the version of record. 

If I were publishing a paper on the methods I am using in my dissertation, I agree that I would be very concerned if that were modified or updated -- if it were, it would no longer be describing what I did. But my code is also very particular to the sources and questions that I am engaging in my dissertation, and as such, that description would be of little use to others as a lesson. As an editor, I have been asking authors to abstract from the particulars of their research methods and to focus on the generalizable problem solving technique. And I think that pedagogical focus is the strength of PH. My concern is that if we don't have some way to keep the lessons pedagogically useful, the publication will suffer.

That being said, perhaps supporting community editing (#304) is beyond the scope of PH because of the expectations of academic journal publishing. I am happy to concede that one project cannot, and should not, be all things to all people.

**Comment by wcaleb** [comment]
Given my impending departure from the team, I don't want to weigh in too heavily here, but I did want to note that I still think we have a promising way forward for satisfying all the above concerns on both sides of the questions this policy raises.

We've already established that we do in fact keep a record of the original lesson embedded within our `git` history. The discussion yesterday on our call clarified for me that @acrymble is most concerned with surfacing that original lesson (perhaps with a designated link to a specific commit representing the author's earliest post-peer-review publication, rather than only a link to the lesson's whole history.) He's less concerned with making the original "pretty print" in a PDF or some other form. If I'm right (and correct me, Adam if I'm wrong), then that fact, combined with some set of author options like those outlined in my text or in Adam's table, may be all we need to satisfy the concerns @acrymble is raising and with minimal additional editorial work in terms of maintaining a "version of record."

As for whether the Project is more like Adam's Number 1 or Number 2, it seems to me that this decision is made more at the point at which we accept lesson pitches and cultivate the lessons (as Matthew and Jeri both note). In other words, the overall drift of the project towards Number 2 won't (IMHO) be fundamentally prevented or changed by adopting and publishing a policy like this. And in the meantime, it doesn't seem to me to hurt our project in any way to try to accommodate the concerns of authors who have already published with us at a past moment when we may have looked more like Number 1 (especially when the author has been as important to the project as @acrymble has been, and when, as outlined above, it looks like it costs us little in terms of additional work). 

In short, if we adopt a flexible policy that lays out clearly the options available to authors regarding ongoing revision, we can rest assured that we are well prepared for the future regardless of whether the project grows in a One-ward or Two-ward direction (a decision made, again, more at the point of lesson recruitment and acceptance than at the point of policy creation).

**Comment by wcaleb** [comment]
P.S. I would add that I think it is important to have a default policy from which authors "opt out," if only for logistical reasons. Given that authors/reviewers sometimes disappear even before publication, we can't always guarantee that we will be able to reach authors down the road and get a firm answer from them about a proposed change that editors feel is vital for keeping the lesson up to date. Ongoing consultation for the life of the lesson may be a good ideal, but I see practical hurdles there that would be easier to manage if we went with a default policy that looks like row 3 of Adam's table (with the added caveats that editors may also choose to deprecate, and that editors will make a good faith effort to notify/consult with the author about changes) but give authors the option of selecting row 1 instead.

**Comment by acrymble** [comment]
I think we can simplify this all by adopting a policy of altering any lesson as we see fit, but rather than claiming these lessons are 'by' the authors, they become 'derived from'. That lets authors maintain their moral rights without attributing new issues to them.

That opens up all of the revision options I know many people would like to see, and protects the rights of authors.

**Comment by acrymble** [comment]
Decisions:

* minor changes of 1-2 sentences worth of change or less can be done under the existing policies.
* If major issues needed, a new ticket opened
* Spanish team consulted if translation already exists.

Also:

* Require a discussion about sustainability policies and defensive writing to minimise needs for change in future #534 
* Require a discussion about deprecation policies #536.

**Comment by mdlincoln** [comment]
Closing this, as developing a statement for our archiving policy is now all contained at #536, and the clarifications to author, editor, and reviewer guidelines needed for sustainability are going to be developed at #534
